### PR TITLE
feat: user defined allocation policy

### PR DESCRIPTION
**Pull request by LexLuthr**
This PR will add the functionality to allow users to set termMin, termMax and expiration for allocations.
It also updated the `get-allocations` command to `list-allocations`

The default are:
TermMin = 180 days
TermMax = 5 years
Expiration = 60 days

### CODE REVIEW

File: cmd/boost/direct_deal.go [hunk 1]
```diff
@@ -38,8 +38,9 @@ var directDealAllocate = &cli.Command{
 		},
 		&cli.StringSliceFlag{
 			Name:     "pieceInfo",
-			Usage:    "data pieceInfo[s] to create the allocation. The format must be pieceCid1=pieceSize1 pieceCid2=pieceSize2",
+			Usage:    "data pieceInfo[s] to create the allocation. The format must be --pieceInfo pieceCid1=pieceSize1 --pieceInfo pieceCid2=pieceSize2",
```
Comment (dirkmc) [inline cmd/boost/direct_deal.go:41]: In general for param names we use snake-case, let's do that for piece-info, term-min and term-max

File: cmd/boost/direct_deal.go [hunk 2]
```diff
@@ -134,6 +152,36 @@ var directDealAllocate = &cli.Command{
...
+			termMin = abi.ChainEpoch(cctx.Int64("termmin"))
+		}
+
+		if cctx.IsSet("termmin") {
+			termMax = abi.ChainEpoch(cctx.Int64("termmax"))
+		}
+
+		if cctx.IsSet("termmin") {
+			exp = abi.ChainEpoch(cctx.Int64("expiration"))
+		}
```
Comment (dirkmc) [inline cmd/boost/direct_deal.go:170]: I'd suggest we set the Value field of the flag to the appropriate constant, eg
```Value=verifregst.MinimumVerifiedAllocationTerm```
Then we can simplify this code to
```
termMin := abi.ChainEpoch(cctx.IsSet("termmin"))
...
```

File: cmd/boost/direct_deal.go [hunk 3]
```diff
@@ -145,9 +179,9 @@ var directDealAllocate = &cli.Command{
 					Provider:   mid,
 					Data:       p.PieceCID,
 					Size:       p.Size,
-					TermMin:    verifregst.MinimumVerifiedAllocationTerm,
-					TermMax:    verifregst.MinimumVerifiedAllocationTerm,
-					Expiration: verifregst.MaximumVerifiedAllocationExpiration,
+					TermMin:    abi.ChainEpoch(cctx.Int64("termmin")),
+					TermMax:    abi.ChainEpoch(cctx.Int64("termmax")),
+					Expiration: abi.ChainEpoch(cctx.Int64("expiration")),
```
Comment (dirkmc) [inline cmd/boost/direct_deal.go:184]: Looks like these also need to be converted to snake-case.
Please test it out before merging.

Comment (LexLuthr) [reply cmd/boost/direct_deal.go]: The basic command works. I am having trouble testing them E2E in devnet. I think Lotus side might be the issue as allocation is never created even when message is sent successfully.

**Review by dirkmc — APPROVED** [review]

File: cmd/boost/direct_deal.go (hunk 3 above)
Comment (dirkmc) [reply cmd/boost/direct_deal.go]: Ok I guess let's merge this PR.

For me it does seem to be working on a devnet:
```
boost$ boost allocate --miner=t01000 --pieceInfo=$COMMP_CID=8388608
about to send message with the following gas costs
max fee:      0.00000000166097256 FIL (absolute maximum amount you are willing to pay to get your transaction confirmed)
gas fee cap:  0.00000000000000012 FIL
gas limit:    13841438
gas premium:  0.000000000000000072 FIL
basefee:      0.0000000000000001 FIL

Proceed? Yes [y] / No [n]:
y
2023-09-05T10:17:07.049+0200	INFO	boost	boost/direct_deal.go:195	submitted data cap allocation message	{"cid": "bafy2bzacedevxmv2zlitcac2u4wbd6kiqlewkice7om24fhd3j52hdof2tcfa"}
2023-09-05T10:17:07.049+0200	INFO	boost	boost/direct_deal.go:196	waiting for message to be included in a block
AllocationID  Client  Miner  PieceCid                                                          PieceSize  TermMin  TermMax  
2             1007    1000   baga6ea4seaqamux3zrxsgljljnlnk45ztx3fxgggfrr3iffhqmniqfaf3ehscaa  8388608    518400   518400   
  Expiration: 172800

$ boost get-allocations
AllocationID  Client  Miner  PieceCid                                                          PieceSize  TermMin  TermMax  
2             1007    1000   baga6ea4seaqamux3zrxsgljljnlnk45ztx3fxgggfrr3iffhqmniqfaf3ehscaa  8388608    518400   518400   
  Expiration: 172800
```

**Pull request merged**